### PR TITLE
Implementation for set_entry_timer extern.

### PR DESCRIPTION
**Pull request by komaljai**
- Implementaion of set_entry_timer extern
- Remove error 'key field larger than previous key, move it before previous key to avoid padding between these keys in EBPF'
- Fix ebpf_packetoffsetbits_save variable declaration.
- Replace NoAction in switch case 0

### CODE REVIEW

File: backends/tc/backend.cpp [hunk 1]
```diff
@@ -446,14 +447,27 @@ void ConvertToBackendIR::postorder(const IR::P4Table *t) {
...
         }
         updateDefaultHitAction(t, tableDefinition);
         updateDefaultMissAction(t, tableDefinition);
         updateMatchType(t, tableDefinition);
         tcPipeline->addTableDefinition(tableDefinition);
+        actionsPerTable.emplace(t, action_list);
     }
 }
 
+const IR::P4Table *ConvertToBackendIR::getTableForAction(cstring action) const {
```
Comment (syogender) [inline backends/tc/backend.cpp]: what will happed is same action is there in multiple table?

There is no way to find action of a table?

File: backends/tc/backend.cpp [hunk 2]
```diff
@@ -446,14 +447,27 @@ void ConvertToBackendIR::postorder(const IR::P4Table *t) {
...
+    for (auto entry : actionsPerTable) {
```
Comment (psivanup) [inline backends/tc/backend.cpp]: If one action is used in multiple tables, then it is duplicated by frontend to make it unique.
So, referring name.originalName will be incorrect. Using toString might be fine as it uses externalName() from Declaration class, please check.

**Review by psivanup — APPROVED** [review]
Looks fine

File: backends/tc/backend.cpp (hunk 2 above)
Comment (komaljai) [reply backends/tc/backend.cpp]: Changed the approch to get table_name

File: backends/tc/backend.cpp (hunk 1 above)
Comment (komaljai) [reply backends/tc/backend.cpp]: It is a valid issue, changed the implementation to get table_name. Getting tableName from ActionTranslationVisitorPNA class, This holds the current table's action being processed.

**Pull request closed without merging**